api/pkg/db/model/model.go

```diff
@@ -21,33 +21,32 @@ import (
 type (
 	Category struct {
 		gorm.Model
-		Name string `gorm:"size:100;not null;unique"`
+		Name string `gorm:"not null;unique"`
 		Tags []Tag
 	}
 
 	Tag struct {
 		gorm.Model
-		Name       string `gorm:"size:100;not null;unique"`
+		Name       string `gorm:"not null;unique"`
 		Category   Category
 		CategoryID uint
 		Resources  []*Resource `gorm:"many2many:resource_tags;"`
 	}
 
 	Catalog struct {
 		gorm.Model
-		Name       string
-		Type       string
-		URL        string
-		Owner      string
+		Name       string `gorm:"primary_key"`
+		Type       string `gorm:"primary_key"`
+		URL        string `gorm:"not null"`
 		ContextDir string
 		Resources  []Resource
-		Revision   string
+		Revision   string `gorm:"not null"`
```

Comment:
Using only `NOT NULL` won't be helpful I guess, we also need `default:null`? WDYT?